examples/metrics/non_stateful.py

```diff
@@ -0,0 +1,62 @@
+# Copyright 2019, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+"""
+This module serves as an example for a simple application using metrics
+Examples show how to recording affects the collection of metrics to be exported
+"""
+import time
+
+from opentelemetry import metrics
+from opentelemetry.sdk.metrics import Counter, Meter
+from opentelemetry.sdk.metrics.export import ConsoleMetricsExporter
+from opentelemetry.sdk.metrics.export.batcher import UngroupedBatcher
+from opentelemetry.sdk.metrics.export.controller import PushController
+
+# Batcher used to collect all created metrics from meter ready for exporting
+# Pass in false for non-stateful batcher. Indicates the batcher computes
+# checkpoints which describe the updates of a single collection period (deltas)
+batcher = UngroupedBatcher(False)
+# Meter is responsible for creating and recording metrics
+meter = Meter(batcher)
+metrics.set_preferred_meter_implementation(meter)
```

Comment:
FYI this example is broken now, possibly as a result of #311.

```
Traceback (most recent call last):
  File "examples/opentelemetry-example-app/src/opentelemetry_example_app/metrics_example.py", line 40, in <module>
    counter.add(25, label_set)
AttributeError: 'DefaultMetric' object has no attribute 'add'
```

[`loader.load_impl`](https://github.com/lzchen/opentelemetry-python/blob/aecc3477b0431821015a0455492764baf90a0c36/opentelemetry-api/src/opentelemetry/util/loader.py#L162-L173) expects `api_type` to be the type of object `factory` produces (i.e. `Meter`) *and* a the concrete type of object to create (i.e. `DefaultMeter`) if the factory is or returns null.

Reply:
Actually it's [metrics_example.py](https://github.com/open-telemetry/opentelemetry-python/pull/341/files#diff-413d5045cc4253f7729fb42b786def46) that's broken for this reason. This example has some other problems:

`set_preferred_meter_implementation` takes a factory, not a meter instance. This only doesn't cause an error here because this example doesn't call `metrics.meter()`, which should fail with a `TypeError`.

This example also creates a `Meter` directly instead of using `metrics.meter()`, which users shouldn't generally do.

Reply:
I added no-op implementations of the metric and handle functions to the default classes so user's who don't have the `meter` loaded would not crash.

I agree the usage should be `set_preferred_metered_implementation(lambda _: Meter())`. However this breaks because `api_type` is `DefaultMeter` [here](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/metrics/__init__.py#L301) but fails the type check in [loader](https://github.com/open-telemetry/opentelemetry-python/blob/master/opentelemetry-api/src/opentelemetry/util/loader.py#L123) because `Meter` is not of type `DefaultMeter`.

I don't want to include too many loader changes in this PR, so I am going to change the code back to use `Meter` as the default implementation just to get the examples running. I think we should fix this in [#394].